src/praisonai/praisonai/bots/_session.py

```
@@ -157,7 +172,31 @@ async def chat(
+            except Exception as exc:
+                # N4: persist the failed inbound message before bubbling.
+                if self._dlq is not None:
+                    try:
+                        # Use run_in_executor to avoid blocking the event loop
+                        await loop.run_in_executor(
+                            None,
+                            lambda: self._dlq.enqueue(
+                                platform=self._platform,
+                                user_id=user_id,
+                                prompt=prompt,
+                                error=f"{type(exc).__name__}: {exc}",
```

Ruff (0.15.12)

[error] Undefined name `exc`

(F821)